astropy/coordinates/transformations/graph.py

```diff
@@ -79,19 +80,13 @@ def __init__(self):
         self._graph = defaultdict(dict)
         self.invalidate_cache()  # generates cache entries
 
-    @property
+    @cached_property
```

Comment:
We do have `astropy.utils.lazyproperty` that is used throughout. It's advantage is that it has default setters and deleters (not possible with `cached_property`) - which slightly simplifies `invalidate_cache`.